src/components/edge.js

```diff
@@ -437,7 +432,11 @@ class Edge extends React.Component<IEdgeProps> {
     // Note: document.getElementById is by far the fastest way to get a node.
     // compare 2.82ms for querySelector('#node-a2 use.node') vs
     // 0.31ms and 99us for document.getElementById()
-    const nodeElem = document.getElementById(`node-${trg[nodeKey]}`);
+    // Although it doesn't allow multiple graphs.
+    // We can use viewWrapperElem to scope the querySelector to a smaller set of elements to improve the speed
+    const nodeElem = viewWrapperElem.querySelector(
+      `[id='node-${trg[nodeKey]}']`
```

Comment:
if we’re doing breaking changes, should we change ids to classes so we don’t need to use these workarounds for duplicate ids within the same document?

Reply:
Perhaps in the next PR we can address that. Please feel free to create a new PR after this if you want.

You are right that this will be a breaking change, so we have the opportunity to make a number of improvements.